authentik/policies/unique_password/models.py

```diff
@@ -0,0 +1,136 @@
+from django.contrib.auth.hashers import identify_hasher
+from django.db import models
+from django.utils.translation import gettext as _
+from rest_framework.serializers import BaseSerializer
+from structlog.stdlib import get_logger
+
+from authentik.core.models import User
+from authentik.policies.models import Policy
+from authentik.policies.types import PolicyRequest, PolicyResult
+from authentik.stages.prompt.stage import PLAN_CONTEXT_PROMPT
+
+LOGGER = get_logger()
+
+
+class UniquePasswordPolicy(Policy):
+    """Policy ensuring a user's password is not identical to a previously used password.
+
+    After enabling the policy, Authentik stores every user's previous password whenever a user
+    changes their own password. Old passwords remain stored in hashed form.
+    """
+
+    password_field = models.TextField(
+        default="password",
+        help_text=_("Field key to check, field keys defined in Prompt stages are available."),
+    )
+
+    # Limit on the number of previous passwords the policy evaluates
+    # Also controls number of old passwords the system stores.
+    num_historical_passwords = models.PositiveIntegerField(
+        default=0,
```

Comment:
What does 0 mean in that case? Also it defaults to 1 in the frontend. If 0 is not valid, we should error on this.